src/components/ImageGrid/ImageGridComponent.tsx

```diff
@@ -98,29 +72,100 @@ const ImageGridComponent = ({
     } else {
       setRowCol({ col: 1, row: 1 });
     }
-  }, [numberOfImgs, images]);
-
-  // to avoid bug undefined url when user change number of images
-  useEffect(() => {
-    for (let i = 0; i < numberOfImgs; i++) {
-      images[i] = exampleImages[i];
-    }
   }, [numberOfImgs]);
 
   return (
     <ImageGrid
       height={imagesGridHeight}
-      width={imagesGridWidth}
+      maxWidth={imagesGridMaxWidth}
       numberOfImgs={numberOfImgs}
       row={rowCol.row}
       col={rowCol.col}
     >
-      {images.map((img, index) => (
-        <ImageWrap>
-          <ImageItem src={img} />
-        </ImageWrap>
-      ))}
+      {numberOfImgs <= 5 &&
+        images
+          .slice(0, numberOfImgs)
+          .map((photo: UnsplashPhotoFortmat, index) => (
+            <ImageWrap>
+              <ImageItem
+                src={photo.url || "default.jpg"}
+                alt={photo?.alt || "photo"}
+              />
+            </ImageWrap>
+          ))}
+
+      {numberOfImgs > 5 &&
+        images.slice(0, 5).map((photo: UnsplashPhotoFortmat, index) => (
+          <ImageWrap style={{ position: "relative" }}>
+            <ImageItem
+              src={photo.url || "default.jpg"}
+              alt={photo?.alt || "photo"}
+            />
+            {index === 4 && (
```

Comment:
following the above comment, this would be:

```suggestion
            {index === MAX_PREVIEW_NUM - 1 && (
```